packages/protocol/contracts/governance/EpochRewards.sol

```diff
@@ -0,0 +1,399 @@
+   * @param rewardsMultiplierMax The max multiplier on target epoch rewards.
+   * @param rewardsMultiplierUnderspendAdjustmentFactor Adjusts the multiplier on target epoch
+   *   rewards when the protocol is running behind the target gold supply.
+   * @param rewardsMultiplierOverspendAdjustmentFactor Adjusts the multiplier on target epoch
+   *   rewards when the protocol is running ahead of the target gold supply.
+   * @param _targetVotingGoldFraction The percentage of floating gold voting to target.
+   * @param _targetValidatorEpochPayment The target validator epoch payment.
+   * @dev Should be called only once.
+   */
+  function initialize(
+    address registryAddress,
+    uint256 targetVotingYieldInitial,
+    uint256 targetVotingYieldMax,
+    uint256 targetVotingYieldAdjustmentFactor,
+    uint256 rewardsMultiplierMax,
+    uint256 rewardsMultiplierUnderspendAdjustmentFactor,
+    uint256 rewardsMultiplierOverspendAdjustmentFactor,
+    uint256 _targetVotingGoldFraction,
+    uint256 _targetValidatorEpochPayment
+  )
+    external
+    initializer
+  {
+    _transferOwnership(msg.sender);
+    setRegistry(registryAddress);
+    setTargetVotingYieldParameters(targetVotingYieldMax, targetVotingYieldAdjustmentFactor);
+    setRewardsMultiplierParameters(
+      rewardsMultiplierMax,
+      rewardsMultiplierUnderspendAdjustmentFactor,
+      rewardsMultiplierOverspendAdjustmentFactor
+    );
+    setTargetVotingGoldFraction(_targetVotingGoldFraction);
+    setTargetValidatorEpochPayment(_targetValidatorEpochPayment);
+    targetVotingYieldParams.target = FixidityLib.wrap(targetVotingYieldInitial);
+    startTime = now;
+  }
+
+  /**
+   * @notice Returns the target voting yield parameters.
+   * @return The target, max, and adjustment factor for target voting yield.
+   */
+  function getTargetVotingYieldParameters() external view returns (uint256, uint256, uint256) {
+    TargetVotingYieldParameters storage params = targetVotingYieldParams;
+    return (params.target.unwrap(), params.max.unwrap(), params.adjustmentFactor.unwrap());
+  }
+
+  /**
+   * @notice Returns the rewards multiplier parameters.
+   * @return The max multiplier and under/over spend adjustment factors.
+   */
+  function getRewardsMultiplierParameters() external view returns (uint256, uint256, uint256) {
+    RewardsMultiplierParameters storage params = rewardsMultiplierParams;
+    return (
+      params.max.unwrap(),
+      params.adjustmentFactors.underspend.unwrap(),
+      params.adjustmentFactors.overspend.unwrap()
+    );
+  }
+
+  /**
+   * @notice Sets the target voting gold fraction.
+   * @param value The percentage of floating gold voting to target.
+   * @return True upon success.
+   */
+  function setTargetVotingGoldFraction(uint256 value) public onlyOwner returns (bool) {
+    require(value != targetVotingGoldFraction.unwrap() && value < FixidityLib.fixed1().unwrap());
+    targetVotingGoldFraction = FixidityLib.wrap(value);
```

Comment:
Consider reordering these lines and using Fixidity comparison operators to reduce `unwrap`s.

Reply:
Will that allow us to compare value to the previous targetVotingGoldFraction? (we will have overwritten it)

Was trying to follow the checks/effects/interactions pattern. Doing it in this order also saves gas in the case where value doesn't change the targetVotingGoldFraction

Reply:
Ah right. I guess the other option would have been to declare a new Fixidity fraction in memory, but that could cost more gas and would definitely add code noise.